### ISSUE TITLE
[REVIEW]: Bempp-cl: A fast Python based just-in-time compiling boundary element library.

### ISSUE THREAD

**Issue by whedon** [openjournals/joss-reviews#2879]
**Submitting author:** @mscroggs (<a href="http://orcid.org/0000-0002-4658-2443">Matthew Scroggs</a>)
**Repository:** <a href="https://github.com/bempp/bempp-cl" target ="_blank">https://github.com/bempp/bempp-cl</a>
**Version:** v0.2.3
**Editor:** @meg-simula
**Reviewer:** @jamtrott, @ramisetti
**Archive:** <a href="https://doi.org/10.5281/zenodo.4618621" target="_blank">10.5281/zenodo.4618621</a>

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

## Status

[![status](https://joss.theoj.org/papers/01ab07c55b10220b61b6e01f7f312920/status.svg)](https://joss.theoj.org/papers/01ab07c55b10220b61b6e01f7f312920)

Status badge code:

```
HTML: <a href="https://joss.theoj.org/papers/01ab07c55b10220b61b6e01f7f312920"><img src="https://joss.theoj.org/papers/01ab07c55b10220b61b6e01f7f312920/status.svg"></a>
Markdown: [![status](https://joss.theoj.org/papers/01ab07c55b10220b61b6e01f7f312920/status.svg)](https://joss.theoj.org/papers/01ab07c55b10220b61b6e01f7f312920)
```
**Reviewers and authors**:

Please avoid lengthy details of difficulties in the review thread. Instead, please create a new issue in the <a href="https://github.com/bempp/bempp-cl" target="_blank">target repository</a> and link to those issues (especially acceptance-blockers) by leaving comments in the review thread below. (For completists: if the target issue tracker is also on GitHub, linking the review thread in the issue or vice versa will create corresponding breadcrumb trails in the link target.)

## Reviewer instructions & questions

@jamtrott & @ramisetti, please carry out your review in this issue by updating the checklist below. If you cannot edit the checklist please:
1. Make sure you're logged in to your GitHub account
2. Be sure to accept the invite at this URL: https://github.com/openjournals/joss-reviews/invitations

The reviewer guidelines are available here: https://joss.readthedocs.io/en/latest/reviewer_guidelines.html. Any questions/concerns please let @meg-simula know.

✨ **Please start on your review when you are able, and be sure to complete your review in the next six weeks, at the very latest** ✨ 

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/bempp/bempp-cl">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@mscroggs) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?
- [x] **Substantial scholarly effort:** Does this submission meet the scope eligibility described in the [JOSS guidelines](https://joss.readthedocs.io/en/latest/submitting.html#substantial-scholarly-effort)

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/bempp/bempp-cl">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@mscroggs) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?
- [x] **Substantial scholarly effort:** Does this submission meet the scope eligibility described in the [JOSS guidelines](https://joss.readthedocs.io/en/latest/submitting.html#substantial-scholarly-effort)

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [ ] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [ ] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

**Comment by whedon** [comment]
Hello human, I'm @whedon, a robot that can help you with some common editorial tasks. @jamtrott, @ramisetti it looks like you're currently assigned to review this paper :tada:.

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

**:star: Important :star:**

If you haven't already, you should seriously consider unsubscribing from GitHub notifications for this (https://github.com/openjournals/joss-reviews) repository. As a reviewer, you're probably currently watching this repository which means for GitHub's default behaviour you will receive notifications (emails) for all reviews 😿

To fix this do the following two things:

1. Set yourself as 'Not watching' https://github.com/openjournals/joss-reviews:

![watching](https://cloud.githubusercontent.com/assets/4483/20250593/64d7ce48-a9de-11e6-9225-d3dfb3e48e68.png)

2. You may also like to change your default settings for this watching repositories in your GitHub profile here: https://github.com/settings/notifications

![notifications](https://cloud.githubusercontent.com/assets/4483/20250604/79eb3144-a9de-11e6-9e51-0fe940966689.png)


For a list of things I can do to help you, just type:

```
@whedon commands
```

For example, to regenerate the paper pdf after making changes in the paper's md or bib files, type:

```
@whedon generate pdf
```

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02879/joss.02879/10.21105.joss.02879.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02879/joss.02879/10.21105.joss.02879.pdf) :page_facing_up: :point_left:

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1016/j.parco.2011.09.001 is OK
- 10.1002/nme.2579 is OK
- 10.1109/TAP.1982.1142818 is OK
- 10.1007/BF01396415 is OK
- 10.1016/j.crma.2004.12.022 is OK
- 10.1016/j.camwa.2017.07.049 is OK
- 10.1145/2833157.2833162 is OK
- 10.1007/978-0-387-68805-3 is OK
- 10.1145/3368618 is OK
- 10.1145/2590830 is OK

MISSING DOIs

- 10.2307/3621632 may be a valid DOI for title: Strongly elliptic systems and boundary integral equations

INVALID DOIs

- None
```

**Comment by meg-simula** [comment]
@jamtrott @ramisetti Thanks again for agreeing to review this submission - I will check back in in 1-2  weeks time to see how matters are progressing. If there is anything in the meanwhile, please let me know.

**Comment by ramisetti** [comment]
@meg-simula I have a problem to review. I don't have access to the reviewer checklist page for this submission.

**Comment by meg-simula** [comment]
@ramisetti: I see. Hm. Did you try these points? 

> 
> If you cannot edit the checklist please:
> 
> Make sure you're logged in to your GitHub account
> Be sure to accept the invite at this URL: https://github.com/openjournals/joss-reviews/invitations

**Comment by ramisetti** [comment]
@meg-simula: I tried as you suggested but I still cannot edit the checklist page.

**Comment by ramisetti** [comment]
Dear @meg-simula,

As I am unable to edit the review checklist, I am providing my review here. Please read below for my review of the _Bempp-cl: A fast Python based just-in-time compiling boundary element library_ package for the **Journal of Open Source Software**.
Conflict of interest
•	I confirm that I have read the JOSS conflict of interest (COI) policy and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.
Code of Conduct
•	I confirm that I read and will adhere to the JOSS code of conduct. 
General checks
•	Repository: Is the source code for this software available at the repository url? Yes
•	License: Does the repository contain a plain-text LICENSE file with the contents of an OSI approved software license? The repository contains LICENSE.md file with the contents of MIT license. The authors should consider renaming it to either LICENSE or COPYING as required for JOSS.
•	Contribution and authorship: Has the submitting author (@mscroggs) made major contributions to the software? Does the full list of paper authors seem appropriate and complete? The submitting authors have made major contributions to the software and the list of the authors seem appropriate and complete as well.
•	Substantial scholarly effort: Does this submission meet the scope eligibility described in the JOSS guidelines. This submission meets the scope eligibility as per the JOSS guidelines and it is clearly visible in terms of the length of commit history, number of commits, LOC, and its potential to attract researchers in this domain.
Functionality
•	 Installation: Does installation proceed as outlined in the documentation? I tried installing the software following instructions for the Quick Install and Installing from source. Both the approaches works smoothly without any issues with Python-3.8 on Mac OS. I have neither tried running it using Docker nor with OpenCL as I have don’t have them.
•	Functionality: Have the functional claims of the software been confirmed? The documentation claims the software can import and export in a number of formats, including Gmsh and VTK but it is not clearly evident from the examples provided in the applications section, except the exporting of gmsh file in the second example. The authors should provide at least one example for each case. Similarly, the authors should provide examples on the CPU and GPU parallelisation. The documentation for the examples should place a phrase related to the functionality to help the users understand the purpose of the example. For example, the documentation for each example can mention a phrase about the functionality in brackets beside the link.
•	Performance: If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.) The authors make no performance claims.
Documentation
•	 A statement of need: Do the authors clearly state what problems the software is designed to solve and who the target audience is? The authors clearly state the type of problems that can be solved using the software. The authors may consider mentioning about the target audience of this software to make it more clear to the potential users
•	 Installation instructions: Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution. The documentation mentions the list of required software to install the software. The dependencies are required to be explicitly installed by the user prior to installing the software. The software lacks the automated package management solution. The authors may consider mentioning the versions of the dependencies particularly for gmsh as this could save time for users in installing the dependencies.
•	Example usage: Do the authors include examples of how to use the software (ideally to solve real-world analysis problems). The documentation includes good number of examples on how to use the software. The authors should fix the broken link for “Weakly imposing a Dirichlet boundary condition” example. I would recommend to separate the tutorials from documentation as it is more intuitive to find examples in tutorials section rather in documentation
•	Functionality documentation: Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)? The python core functionality is available, but I believe it is not up to a satisfactory level to be useful for the prospective software developers interested in the software
•	 Automated tests: Are there automated tests or manual steps described so that the functionality of the software can be verified? Although the repository contains a list of unit test but the documentation does not mention anything about any automated tests. The authors should consider updating the documentation with information on unit tests.
•	 Community guidelines: Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support: Information on third party contribution to the software is not provided. The documentation provides information on how to reporting issues or problems through GitHub repository. It also provides an email to seek any support related to the software.
Software paper
•	 Summary: Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided? The paper tries to give a high-level functionality description through good references to BEM. I believe this will help non-specialist readers as well.
•	 A statement of need: Do the authors clearly state what problems the software is designed to solve and who the target audience is? The authors clearly state the type of problems that can be solved using the software. The authors may consider mentioning about the target audience of this software to make it more clear to readers
•	 State of the field: Do the authors describe how this software compares to other commonly-used packages? This submission lacks information on why this software is better and how this compares to any of the existing open source software. The authors should consider including information in the paper if there are any existing other opensource software and how this software compares with them. 
•	Quality of writing: Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)? The paper is well written and structured as per the JOSS guidelines. However, there are few minor mistakes that should be corrected to make it better. i) “Calder’on” should be changed to “Calderón” in Section: Statement of need ii) “writing” should be changed to “written” in Section: Discretisation and solvers, iii) “fiels” should be changed to “field” in Section: Potential and far field operators, and iv) “in in” should be changed to “in” in Section: Further information
•	References: Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper citation syntax? I think the list of references is complete and everything is cited appropriately

**Comment by meg-simula** [comment]
Thanks @ramisetti for the prompt review! @mscroggs Are you able to handle the review comments in the format listed above?

**Comment by meg-simula** [comment]
@arfon Any ideas on why @ramisetti is not assigned to this issue (and not allowed to edit the checkboxes)? Anything I can do regarding this?

**Comment by mscroggs** [comment]
Yes, this format is fine. I'll get some revisions done soon

**Comment by arfon** [comment]
@whedon re-invite @ramisetti as reviewer

**Comment by whedon** [comment]
OK, the reviewer has been re-invited.

@ramisetti please accept the invite by clicking this link: https://github.com/openjournals/joss-reviews/invitations

**Comment by arfon** [comment]
> @arfon Any ideas on why @ramisetti is not assigned to this issue (and not allowed to edit the checkboxes)? Anything I can do regarding this?

@meg-simula - looks like something went awry with the repository invite. Can you try clicking the above link again @ramisetti? The invite should be shown at the top of this page: https://github.com/openjournals/joss-reviews/invitations

**Comment by ramisetti** [comment]
Thank you @arfon and @meg-simula. I can now edit the checkboxes.

**Comment by meg-simula** [comment]
> @whedon re-invite @ramisetti as reviewer

Thanks @arfon. Next time this happens, should I just try this re-invite?

**Comment by arfon** [comment]
> Thanks @arfon. Next time this happens, should I just try this re-invite?

Yes, that's it!

**Comment by whedon** [comment]
:wave: @ramisetti, please update us on how your review is going.

**Comment by meg-simula** [comment]
I think this was an automatic comment from @whedon, please feel free to disregard @ramisetti

**Comment by meg-simula** [comment]
@mscroggs Could you please give an update on the status of the review process?

**Comment by mscroggs** [comment]
@meg-simula We're waiting for me to finish making changes, they should be done shortly

**Comment by mscroggs** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02879/joss.02879/10.21105.joss.02879.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02879/joss.02879/10.21105.joss.02879.pdf) :page_facing_up: :point_left:

**Comment by mscroggs** [comment]
I've updated the paper following the reviewer comments. Notes on changes made can be found in this issue: https://github.com/bempp/bempp-cl/issues/94

**Comment by mscroggs** [comment]
@ramisetti @jamtrott Thanks for the reviews. I've just finished making changes to the paper and documentation (summary of changes at bempp/bempp-cl#94)

**Comment by meg-simula** [comment]
Thanks @mscroggs, well done! 

So, it looks like all the boxes are checked, except one regarding State of the art. @ramisetti is this still an issue?

@jamtrott @ramisetti Are you satisfied with the revisions and changes made by @mscroggs, and have all your points been addressed?

**Comment by ramisetti** [comment]
@meg-simula  I am satisfied with the revisions made by @mscroggs. I believe the State of the art may provide a better comparison of the software to other existing ones. The authors may mention a phrase on how Bempp-cl is better/different from other any existing ones.

**Comment by jamtrott** [comment]
Hi!

Well done, @mscroggs. I'm happy with the changes, and I think the updated version is a nice improvement.

I don't have any further issues, but I will point out a couple of minor nitpicks.

In the section "Statement of need":
  - "The syntax of library" should perhaps be "The syntax of the library", or maybe "The syntax used by the library", or something.
  -  "[...], making implementing a problem [...]" sounds a bit awkward, and could perhaps be rephrased.
  - In the next paragraph, I find myself a bit confused by the use of parentheses. In particular, "[...] and (abem for acoustics) (Jargstorff, n.d.)", and "PyGBe (for biomolecular electrostatics) (Cooper et al., 2016)". These look a little strange to me, and perhaps it reads better if the parentheses are simply removed.


Once more, good work.

Sorry, it took a while for me to respond.

**Comment by mscroggs** [comment]
Thanks both, I'll get these sorted today

**Comment by mscroggs** [comment]
@whedon generate pdf

**Comment by mscroggs** [comment]
@jamtrott @ramisetti I've fixed those issues and expanded the statement of need a little. You can view a diff of the changes here: bempp/bempp-cl#105

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02879/joss.02879/10.21105.joss.02879.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02879/joss.02879/10.21105.joss.02879.pdf) :page_facing_up: :point_left:

**Comment by ramisetti** [comment]
@meg-simula @mscroggs I am happy with the revised manuscript. I think all my points have been now addressed.

**Comment by danielskatz** [comment]
👋 @meg-simula - I think this is waiting on you at this point

**Comment by meg-simula** [comment]
@whedon generate pdf

**Comment by meg-simula** [comment]
@whedon check references

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02879/joss.02879/10.21105.joss.02879.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02879/joss.02879/10.21105.joss.02879.pdf) :page_facing_up: :point_left:

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1016/j.parco.2011.09.001 is OK
- 10.1002/nme.2579 is OK
- 10.1109/TAP.1982.1142818 is OK
- 10.1007/BF01396415 is OK
- 10.1016/j.crma.2004.12.022 is OK
- 10.1016/j.camwa.2017.07.049 is OK
- 10.1145/2833157.2833162 is OK
- 10.1007/978-0-387-68805-3 is OK
- 10.1145/3368618 is OK
- 10.1145/2590830 is OK
- 10.21105/joss.00043 is OK
- 10.1007/978-3-642-23099-8 is OK

MISSING DOIs

- 10.2307/3621632 may be a valid DOI for title: Strongly elliptic systems and boundary integral equations

INVALID DOIs

- None
```

**Comment by meg-simula** [comment]
Thanks again @jamtrott and @ramisetti for completing the reviews - much appreciated - and @mscroggs for addressing all comments. This looks good to go from my side. (The paper might be a bit long for the current JOSS standard, but I think the manuscript is valuable as is, and would recommend its current form.)

**Comment by meg-simula** [comment]
At this point could you:
- [ ] Make a tagged release of your software, and list the version tag of the archived version here.
- [ ] Archive the reviewed software in Zenodo or a similar service (e.g., figshare, an institutional repository)
- [ ] Check the archival deposit (e.g., in Zenodo) has the correct metadata. This includes the title (should match the paper title) and author list (make sure the list is correct and people who only made a small fix are not on it). You may also add the authors' ORCID.
- [ ] Please list the DOI of the archived version here.

I can then move forward with accepting the submission.

**Comment by meg-simula** [comment]
PS: whedon check references highlighted one possibly erraneous DOI in the references, would you @mscroggs check that prior to the above, please?

**Comment by mscroggs** [comment]
The missing DOI is for an old book that I couldn't find a DOI for. The suggested DOI is for a review of it, not the book itself. I'll sort those tasks out now

**Comment by meg-simula** [comment]
Ok, sounds good.
--
Marie

**Comment by mscroggs** [comment]
Tagged release: https://github.com/bempp/bempp-cl/releases/tag/v0.2.3

**Comment by mscroggs** [comment]
@meg-simula: I think those tasks are all done now. Thanks for editing.

And thanks @jamtrott and @ramisetti for helpful reviewer comments.

**Comment by meg-simula** [comment]
@whedon set 10.5281/zenodo.4618621 as archive

**Comment by whedon** [comment]
OK. <a href="https://doi.org/10.5281/zenodo.4618621" target="_blank">10.5281/zenodo.4618621</a> is the archive.

**Comment by meg-simula** [comment]
@whedon set v0.2.3 as version

**Comment by whedon** [comment]
OK. v0.2.3 is the version.

**Comment by meg-simula** [comment]
@whedon accept

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1016/j.parco.2011.09.001 is OK
- 10.1002/nme.2579 is OK
- 10.1109/TAP.1982.1142818 is OK
- 10.1007/BF01396415 is OK
- 10.1016/j.crma.2004.12.022 is OK
- 10.1016/j.camwa.2017.07.049 is OK
- 10.1145/2833157.2833162 is OK
- 10.1007/978-0-387-68805-3 is OK
- 10.1145/3368618 is OK
- 10.1145/2590830 is OK
- 10.21105/joss.00043 is OK
- 10.1007/978-3-642-23099-8 is OK

MISSING DOIs

- 10.2307/3621632 may be a valid DOI for title: Strongly elliptic systems and boundary integral equations

INVALID DOIs

- None
```

**Comment by whedon** [comment]
:wave: @openjournals/joss-eics, this paper is ready to be accepted and published.

 Check final proof :point_right: https://github.com/openjournals/joss-papers/pull/2153

If the paper PDF and Crossref deposit XML look good in https://github.com/openjournals/joss-papers/pull/2153, then you can now move forward with accepting the submission by compiling again with the flag `deposit=true` e.g.
 ```
@whedon accept deposit=true
```

**Comment by arfon** [comment]
@whedon accept deposit=true

**Comment by whedon** [comment]
```
Doing it live! Attempting automated processing of paper acceptance...
```

**Comment by whedon** [comment]
🐦🐦🐦 👉 [Tweet for this paper](https://twitter.com/JOSS_TheOJ/status/1372800598702059522) 👈 🐦🐦🐦

**Comment by whedon** [comment]
🚨🚨🚨 **THIS IS NOT A DRILL, YOU HAVE JUST ACCEPTED A PAPER INTO JOSS!** 🚨🚨🚨

 Here's what you must now do:

0. Check final PDF and Crossref metadata that was deposited :point_right: https://github.com/openjournals/joss-papers/pull/2155
1. Wait a couple of minutes to verify that the paper DOI resolves [https://doi.org/10.21105/joss.02879](https://doi.org/10.21105/joss.02879)
2. If everything looks good, then close this review issue.
3. Party like you just published a paper! 🎉🌈🦄💃👻🤘

 Any issues? Notify your editorial technical team...

**Comment by arfon** [comment]
@jamtrott, @ramisetti - many thanks for your reviews here and to @meg-simula for editing this submission. JOSS relies upon the volunteer efforts of people like you all, and we simply wouldn't be able to do this without you ✨ 

@mscroggs - your paper is now accepted and published in JOSS :zap::rocket::boom:

**Comment by whedon** [comment]
:tada::tada::tada: Congratulations on your paper acceptance! :tada::tada::tada:

If you would like to include a link to your paper from your README use the following code snippets:

```
Markdown:
[![DOI](https://joss.theoj.org/papers/10.21105/joss.02879/status.svg)](https://doi.org/10.21105/joss.02879)

HTML:
<a style="border-width:0" href="https://doi.org/10.21105/joss.02879">
  <img src="https://joss.theoj.org/papers/10.21105/joss.02879/status.svg" alt="DOI badge" >
</a>

reStructuredText:
.. image:: https://joss.theoj.org/papers/10.21105/joss.02879/status.svg
   :target: https://doi.org/10.21105/joss.02879
```
This is how it will look in your documentation:

[![DOI](https://joss.theoj.org/papers/10.21105/joss.02879/status.svg)](https://doi.org/10.21105/joss.02879)

**We need your help!**

Journal of Open Source Software is a community-run journal and relies upon volunteer effort. If you'd like to support us please consider doing either one (or both) of the the following:

- Volunteering to review for us sometime in the future. You can add your name to the reviewer list here: https://joss.theoj.org/reviewer-signup.html
- Making a small donation to support our running costs here: https://numfocus.org/donate-to-joss
